From 7dac334cf1ce4507718b00ef9c9ba0f26d8eb965 Mon Sep 17 00:00:00 2001
From: Michael Webster <miketwebster@gmail.com>
Date: Thu, 11 Aug 2022 10:11:55 -0400
Subject: [PATCH] windowManager.js: Complete the workspace switch immediately
 if no windows end up being animated.

Completion was dependent upon at least one window being involved
in the transition, so when no windows were visible, it was leaving
the compositor in an undefined state.

Fixes #11005.
---
 js/ui/windowManager.js | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index 96075ad883..a6eec7ae63 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -1136,7 +1136,8 @@ var WindowManager = class WindowManager {
             // Muffin 5.2 window.showing_on_its_workspace() no longer
             // ends up filtering the desktop window (If I re-add it, it
             // breaks things elsewhere that rely on the new behavior).
-            if (meta_window.get_window_type() === Meta.WindowType.DESKTOP) {
+            if (meta_window.get_window_type() === Meta.WindowType.DESKTOP ||
+                meta_window.get_window_type() === Meta.WindowType.OVERRIDE_OTHER) {
                 continue;
             }
 
@@ -1185,6 +1186,11 @@ var WindowManager = class WindowManager {
             }
         }
 
+        if (to_windows.size === 0 && from_windows.size === 0) {
+            this._cinnamonwm.completed_switch_workspace();
+            return;
+        }
+
         kill_id = this._cinnamonwm.connect('kill-switch-workspace', cinnamonwm => {
             let iter = to_windows.forEach((actor) => {
                 cleanup_window_effect(actor);
